scripts/test-e2e-local.js

```diff
@@ -250,4 +272,34 @@ if (argv.target === 'RNTester') {
   popd();
 }
 
-exit(0);
+async function main() {
+  /*
+   * see the test-local-e2e.js script for clean up process
+   */
+
+  // command order: we ask the user to select if they want to test RN tester
+  // or RNTestProject
+
+  // if they select RN tester, we ask if iOS or Android, and then we run the tests
+  // if they select RNTestProject, we run the RNTestProject test
+
+  checkPackagerRunning();
+
+  const branchName = exec('git rev-parse --abbrev-ref HEAD', {
+    silent: true,
+  }).stdout.trim();
+  const onReleaseBranch = branchName.endsWith('-stable');
+
+  let circleCIArtifacts = await setupCircleCIArtifacts(
+    argv.circleciToken,
+    branchName,
+  );
```

Comment:
I think it might be realistic to only allow for the circleCI flow to only work on release branches, on main it will almost always be in a "in progress" situation which ends up with this non-very-graceful error:
<img width="919" alt="Screenshot 2023-07-05 at 14 08 26" src="https://github.com/facebook/react-native/assets/16104054/33734997-db91-4aad-9b5e-759d26e3489c">

Reply:
uhm.. I prefer to fail with a proper message and have the user type the command again, removing the token, so that going to the old flow is intentional.
I don't want to automatically fallback to the old flow: the expectations could be disappointed if the user misses the error message and the log is quite verbose, so it's easy to miss the error message if we don't interrupt the execution.